services/gitdiff/gitdiff.go

```diff
@@ -382,8 +382,8 @@ func (diffFile *DiffFile) GetType() int {
 }
 
 // GetTailSection creates a fake DiffLineSection if the last section is not the end of the file
-func (diffFile *DiffFile) GetTailSection(gitRepo *git.Repository, leftCommit, rightCommit *git.Commit) *DiffSection {
-	if len(diffFile.Sections) == 0 || diffFile.Type != DiffFileChange || diffFile.IsBin || diffFile.IsLFSFile {
+func (diffFile *DiffFile) GetTailSection(leftCommit, rightCommit *git.Commit) *DiffSection {
+	if len(diffFile.Sections) == 0 || leftCommit == nil || diffFile.Type != DiffFileChange || diffFile.IsBin || diffFile.IsLFSFile {
```

Comment:
Wait, that seems strange:
What does the commit being missing have to do with whether we have to create the `expand` button?
These two things seem unrelated to me…

Reply:
And if we do need it, wouldn't we need the check `rightCommit == null` as well?

Reply:
It's not strange, it doesn't need `leftCommit == nil` because `DiffFileChange` already implies `leftCommit == nill`.

This change just clarifies it.

And, `rightCommit` can never be nil